src/txn/txn.c

```diff
@@ -196,13 +196,13 @@ __wt_txn_update_oldest(WT_SESSION_IMPL *session)
 	oldest_session = NULL;
 	prev_oldest_id = txn_global->oldest_id;
 
-	/* For pure read-only workloads, avoid scanning. */
-	if (prev_oldest_id == current_id) {
-		/* Check that the oldest ID has not moved in the meantime. */
-		if (prev_oldest_id == txn_global->oldest_id &&
-		    txn_global->scan_count == 0)
```

Comment:
This clause would have returned early before if ```(force && scan_count == 0)```. What was the reasoning behind change that?